### ISSUE TITLE
Improve developer experience with finishIfInvalidAuth()

### ISSUE THREAD

**Issue by chris-mitchell** [dropbox/kaiken#19]
Some users are forgetting to call `finishIfInvalidAuth()` in their `onAttach` method in their fragments. 

Since the users call it in their Activity, this usually works in development.  But when the Activity is destroyed, the Fragment can be recreated during the `super.onCreate()` call in the Activity.  This causes a crash when the Fragment tries to call `inject()`. 

It would be good to either add a lint rule to check for this, or see if there is a way to remove the need to call `finishIfInvalidAuth()` in the Fragment.

**Comment by rodgavila** [comment]
Yes. I think we could a lint rule that checks that all fragments that implement `AuthRequiredFragment` or `AuthOptionalFragment` should always call `finishIfInvalidAuth()` on their `onAttach()` method. `AuthAwareFragment` should be safe, because they always rely on their parent `Fragment` / `Activity`.

We might also want to add concrete class versions of all of the base interfaces. However, this would violate one of our Guiding Principles: _composition over inheritance_. But I can see how there's people that actually want to make it as part of their fragment / activities tree to avoid having to call `finishIfInvalidAuth()` all the time.

**Comment by digitalbuddha** [comment]
Since already a processor might as well fail during compilation than lint
